pkg/policy/controller.go

```diff
@@ -159,9 +160,21 @@ func (pc *PolicyController) addPolicy(obj interface{}) {
 	// policy.spec.background -> "True"
 	// register with policy meta-store
 	pc.pMetaStore.Register(*p)
-	if !p.Spec.Background {
+	if p.Spec.Background == nil {
+		// skipped policy mutation default -> do not process
 		return
 	}
+
+	if !*p.Spec.Background {
+		return
+	}
+	// If userInfo is used then skip the policy
+	// ideally this should be handled by background flag only
+	if err := policy.ContainsUserInfo(*p); err != nil {
+		// contains userInfo used in policy
+		return
+	}
```

Comment:
I don't see anywhere we enqueue the policy when it has no userInfo. 
The condition for such a scenario should be `p.Spec.Background == nil` && `!policy.ContainsUserInfo(*p)`.

Reply:
Updated as suggested, also created issue https://github.com/nirmata/kyverno/issues/598 for discussion on background processing.